airbyte-db/db-lib/src/main/java/io/airbyte/db/jdbc/StreamingJdbcDatabase.java

```diff
@@ -69,7 +69,9 @@ public <T> Stream<T> unsafeQuery(final CheckedFunction<Connection, PreparedState
       return toUnsafeStream(statement.executeQuery(), recordTransform, streamingConfig)
           .onClose(() -> {
             try {
-              connection.setAutoCommit(true);
+              if (!connection.getAutoCommit()) {
+                connection.setAutoCommit(true);
```

Comment:
I not familiar with the intricacies here
But Is this logically any different ?
in both before and after we end with `autoCommit == true`

Reply:
Logically it could be different if we had multiple transactions within a single streaming query which mutate the source DB (which we don't)

The setting autocommit = true was just done to deal with a quirk in the POstgres JDBC driver + fetch size interaction which does not apply for mysql.